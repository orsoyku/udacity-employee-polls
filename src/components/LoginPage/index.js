import React, { useEffect } from "react";
import AuthForm from "../AuthForm/index";
import { useSelector } from "react-redux";
import { useNavigate } from "react-router-dom";

const LoginPage = () => {
  const navigate = useNavigate();
  const auth = useSelector((state) => state.loginReducer);
  useEffect(() => {
    auth.isAuthenticated ? navigate("/") : navigate("/login");
  }, [auth.isAuthenticated]);
  return (
    <React.Fragment>
      <AuthForm />
    </React.Fragment>
  );
};

export default LoginPage;
