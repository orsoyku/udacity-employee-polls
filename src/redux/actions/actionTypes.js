export const GET_USERS_SUCCESS = "GET_USERS_SUCCESS";
export const LOGIN_REQUEST_CHECK = "LOGIN_REQUEST_CHECK";
export const CREATE_QUESTION = "CREATE_QUESTION";
export const GET_QUESTIONS = "GET_QUESTIONS";
export const ANSWER_QUESTION = "ANSWER_QUESTION";
export const ADD_QUESTION_TO_USER = "ADD_QUESTION_TO_USER";
export const ADD_ANSWER_TO_USER = "ADD_ANSWER_TO_USER";
export const LOG_OUT = "LOG_OUT";
