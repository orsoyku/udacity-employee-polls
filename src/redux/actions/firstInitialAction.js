import * as actionTypes from "./actionTypes";
import { _getUsers, _getQuestions } from "../../utils/data";

export const getAllInitialData = () => (dispatch) => {
  _getUsers().then((response) => {
    dispatch({ type: actionTypes.GET_USERS_SUCCESS, payload: response });
  });
  _getQuestions().then((response) => {
    dispatch({ type: actionTypes.GET_QUESTIONS, payload: response });
  });
};
