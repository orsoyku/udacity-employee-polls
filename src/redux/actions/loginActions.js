import * as actionTypes from "./actionTypes";
import { users } from "../../utils/data";

export const login = (creds) => (dispatch) => {
  const { name, password } = creds;
  if (Object.keys(users).includes(name)) {
    const user = users[name];
    if (user.password === password) {
      alert("login is success");
      dispatch({
        type: actionTypes.LOGIN_REQUEST_SUCCESS,
        payload: { isAuthenticated: true, id: user.id },
      });
    }
    else{
      alert("password or username is wrong..")
    }
  }
};
