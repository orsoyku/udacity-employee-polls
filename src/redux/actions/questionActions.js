import * as actionTypes from "./actionTypes";
import { _saveQuestion, _saveQuestionAnswer } from "../../utils/data";

export const createQuestion = (questionInfo) => (dispatch) => {
  _saveQuestion(questionInfo).then((response) => {
    dispatch({
      type: actionTypes.CREATE_QUESTION,
      payload: response,
    });
    dispatch({
      type: actionTypes.ADD_QUESTION_TO_USER,
      payload: response,
    });
  });
};

export const saveQuestionAnswer = (authedUser, qid, answer) => (dispatch) => {
  if (authedUser && qid && answer) {
    _saveQuestionAnswer({ authedUser, qid, answer }).then(() => {
      dispatch({
        type: actionTypes.ANSWER_QUESTION,
        payload: { authedUser, qid, answer },
      });
      dispatch({
        type: actionTypes.ADD_ANSWER_TO_USER,
        payload: { authedUser, qid, answer },
      });
    });
  }
};
