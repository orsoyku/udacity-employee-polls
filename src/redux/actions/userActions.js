import * as actionTypes from "./actionTypes";

export const addQuestionToUser = (question) => {
  return { type: actionTypes.ADD_QUESTION_TO_USER, payload: question };
};

export const addAnswerToUser = (question) => {
  return { type: actionTypes.ADD_ANSWER_TO_USER, payload: question };
};
