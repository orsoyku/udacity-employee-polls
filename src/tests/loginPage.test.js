import { Provider } from "react-redux";
import { createStore } from "redux";
import LoginPage from "../components/LoginPage";
import { getAllInitialData } from "../redux/actions/firstInitialAction";
import { render, fireEvent } from "@testing-library/react";
import userEvent from '@testing-library/user-event';
import configureStore from "../redux/reducers/configureStore";
import { BrowserRouter } from "react-router-dom";

describe("Login page testing", () => {
  //snapshot testing
  it("Snapshot test when login componet was created", () => {
    const store = configureStore();
    const loginPage = render(
      <Provider store={store}>
        <BrowserRouter>
          <LoginPage />
        </BrowserRouter>
      </Provider>
    );
    expect(loginPage).toMatchSnapshot();
  });

  it("Ä°f selectbox value is not chosen", () => {
    const store = configureStore();
    const loginPage = render(
      <Provider store={store}>
        <BrowserRouter>
          <LoginPage />
        </BrowserRouter>
      </Provider>
    );
    const cred = loginPage.getByTestId("cred");
    const submitButton = loginPage.getByTestId("submitButton");
    fireEvent.change(cred, { target: { value: '' } });
    expect(submitButton).toBeDisabled();
  });

});
