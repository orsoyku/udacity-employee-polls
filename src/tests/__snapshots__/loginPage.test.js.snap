// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page testing Snapshot test when login componet was created 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="authForm"
      >
        <div
          class="container"
        >
          <div
            class="row d-flex justify-content-center"
          >
            <div
              class="col-md-4"
            >
              <form
                id="loginform"
              >
                <div
                  class="form-group"
                >
                  <select
                    class="form-select form-select-lg"
                    data-testid="cred"
                    name="cred"
                  >
                    <option
                      value=""
                    >
                      Please select a user...
                    </option>
                  </select>
                </div>
                <button
                  class="btn btn-primary mt-2"
                  data-testid="submitButton"
                  disabled=""
                  type="submit"
                >
                  Submit
                </button>
              </form>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="authForm"
    >
      <div
        class="container"
      >
        <div
          class="row d-flex justify-content-center"
        >
          <div
            class="col-md-4"
          >
            <form
              id="loginform"
            >
              <div
                class="form-group"
              >
                <select
                  class="form-select form-select-lg"
                  data-testid="cred"
                  name="cred"
                >
                  <option
                    value=""
                  >
                    Please select a user...
                  </option>
                </select>
              </div>
              <button
                class="btn btn-primary mt-2"
                data-testid="submitButton"
                disabled=""
                type="submit"
              >
                Submit
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
